bundles/mapping/mapmyplaces/domain/MyPlacesLayerModelBuilder.js

```diff
@@ -40,6 +41,18 @@ Oskari.clazz.define('Oskari.mapframework.bundle.mapmyplaces.domain.MyPlacesLayer
                     maplayerService.addLayerGroup(Oskari.clazz.create('Oskari.mapframework.domain.MaplayerGroup', group));
                 }
             }
+            if (!this.dataProviderId) {
+                this.dataProviderId = 'myplaces';
```

Comment:
Create the id with the same logic as group since any negative ids are handled differently (edit tool not shown etc). We could probably just use the groupId as is for dataprovider id as well.